src/main/scala/corpus.scala

```diff
@@ -0,0 +1,87 @@
+package treesitter.lsp
+
+import parsley.character.*
+import parsley.debug.*
+import parsley.combinator.*
+import parsley.Parsley
+
+enum LispNode:
+  case Leaf(name: String)
+  case Nest(title: String, nodes: List[LispNode])
+
+object LispNode:
+  export parsers.parser
+
+  private object parsers:
+    def constructNode(nm: String, subnodes: List[LispNode]) =
```

Comment:
I'd make this an `apply` of `LispNode`'s companion object. Reason being, you can mix in `parsley.genericbridges.ParserBridge2[String, List[LispNode], LispNode]` to the companion object and get a free `apply` that takes parsers as arguments instead of the values (see comment where you used this function)